packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/sdk/test/arm-test/src/api/networkAnalyticsApiContext.ts

```diff
@@ -65,3 +71,28 @@ export function createNetworkAnalyticsApi(
     subscriptionId,
   } as NetworkAnalyticsApiContext;
 }
+
+/** Get the ARM endpoint for the client. */
+function getArmEndpoint(cloudSetting?: string): string | undefined {
+  if (cloudSetting === undefined) {
+    return undefined;
+  }
+  if (cloudSetting === "AZURE_CHINA_CLOUD") {
+    return "https://management.chinacloudapi.cn/";
+  } else if (cloudSetting === "AZURE_US_GOVERNMENT")
+    return "https://management.usgovcloudapi.net/";
+  else if (cloudSetting === "AZURE_GERMAN_CLOUD") {
+    return "https://management.microsoftazure.de/";
+  } else if (cloudSetting === "AZURE_PUBLIC_CLOUD") {
+    return "https://management.azure.com/";
+  } else {
+    throw new Error("Unknown cloud setting: " + cloudSetting);
+  }
+}
+
+export enum KnownAzureCloud {
```

Comment:
@xiangyan99 I am looking at the demo pr and I wonder why we proposed to [use a helper function to convert from enum to endpoint](https://github.com/Azure/azure-sdk-for-js/pull/31611/files#diff-591bff0b58077c9dfbd150b42bba1ae0f0ee3689f1f38a50f3ee38ff994af82cR159-R178), why not directly using a enum to save the endpoint directly like below?

```ts
export enum KnownAzureCloud {
    AZURE_CHINA_CLOUD = "https://management.chinacloudapi.cn/",
}
```